framer/Animation.coffee

```diff
@@ -292,6 +296,24 @@ class exports.Animation extends BaseClass
 	_updateNumberValue: (key, value) =>
 		@_target[key] = Utils.mapRange(value, 0, 1, @_stateA[key], @_stateB[key])
 
+	_updateNumericObjectValue: (propKeys, key, value) =>
+		valueA = @_stateA[key]
+		valueB = @_stateB[key]
+
+		result = {}
+
+		for propKey in propKeys
+			keyValueA = if _.isNumber(valueA) then valueA else valueA[propKey]
+			keyValueB = if _.isNumber(valueB) then valueB else valueB[propKey]
+			keyValueA ?= keyValueB
+			keyValueB ?= keyValueA
+			result[propKey] = Utils.mapRange(value, 0, 1, keyValueA, keyValueB)
+
+		if _.uniq(_.values(result)).length is 1
+			result = result[propKeys[0]]
```

Comment:
There's a lot happening here that isn't directly clear. Maybe the naming of variables could be better, maybe it needs some comments.